### ISSUE TITLE
Filter on headers should check whether values are valid or not  before setting headers

### ISSUE THREAD

**Issue by dreamdevil00** [mean-expert-official/loopback-sdk-builder#520]
#### What type of issue are you creating?
- [x] Bug
- [ ] Enhancement
- [ ] Question

#### What version of this module are you using?
- [ ] 2.0.10 (Stable)
- [x] 2.1.0-rc.n (2.1 Release Candidate n)
- [ ] Other

Write other if any:

#### Please add a description for your issue:

I have to get infomation count with where filter, and the filter is
```
{
  "and": [{
    "or": [{
        "yearno": {
          "like": "2010",
          "options": "i"
        }
      },
      {
        "siteName": {
          "like": "2010",
          "options": "i"
        }
      }
    ]
  }]
}
```

it turns out one url 
```
demoApi/count?where[and][0][or][0][yearno][like]=2010&where[and][0][or][0][yearno][options]=i&where[and][0][or][1][siteName][like]=2010&where[and][0][or][1][siteName][options]=i
```

which does not work at all

Instead, I think the url should be like this
```
demoApi/count?where={"and":[{"or":[{"yearno":{"like":"2010","options":"i"}},{"siteName":{"like":"2010","options":"i"}}]}]}
```

Please confirm

**Comment by dreamdevil00** [comment]
Finally, I found the problem.
This problem occoured when the the search value is Chinese.
```
Failed to execute 'setRequestHeader' on 'XMLHttpRequest': '{"where":{"and":[{"or":[{"pname":"2010年四项目"}]}]},"limit":20,"offset":0,"order":""}' is not a valid HTTP header field value."
```
I found the default behavior of filter is filter on headers in lb.config.ts
When I changed
```
private static filterOn: string = 'headers';
```
to 
```
private static filterOn: string = 'url';
```
the problem gone.

I found that it does not check the value before setting headers, [the source code](https://github.com/mean-expert-official/loopback-sdk-builder/blob/ae840a90694d63bf226a1e335df17069d941d310/lib/angular2/shared/services/core/base.ejs#L84), this leads to  the problem.
